### PR TITLE
boards: cpu: replace GPIO_UNDEF by SPI_CS_UNDEF in SPI configuration

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

STM32, Kinetis and cc2538 based boards are updated.

This PR make the SPI implementations and configurations of kinetis, stm32 and cc2538 cpu/boards valid API wise.

It's a takeover of #6486

### Testing procedure

A green CI should be enough

### Issues/PRs references

Takeover of #6486, so closes #6486

### CODE REVIEW

File: cpu/kinetis/periph/spi.c [hunk 1]
```diff
@@ -136,7 +136,7 @@ int spi_init_cs(spi_t bus, spi_cs_t cs)
     }
     else {
         if ((cs >= SPI_HWCS_NUMOF) ||
-            (spi_config[bus].pin_cs[cs] == GPIO_UNDEF)) {
+            (spi_config[bus].pin_cs[cs] == SPI_CS_UNDEF)) {
```
Comment (gschorcht) [inline cpu/kinetis/periph/spi.c:139]:
```suggestion
            (gpio_is_valid(spi_config[bus].pin_cs[cs])) {
```

File: cpu/stm32/include/periph/cpu_spi.h [hunk 2]
```diff
@@ -48,6 +48,21 @@ extern "C" {
  */
 #define SPI_HWCS(x)         (SPI_HWCS_MASK | x)
 
+/**
+ * @brief   Define value for unused CS line
+ */
+#define SPI_CS_UNDEF    (GPIO_UNDEF)
```
Comment (gschorcht) [inline cpu/stm32/include/periph/cpu_spi.h:54]:
```suggestion
#define SPI_CS_UNDEF        (GPIO_UNDEF)
```
At will, just to be aligned with the definition above.

File: cpu/stm32/periph/spi.c [hunk 3]
```diff
@@ -172,7 +172,7 @@ int spi_init_cs(spi_t bus, spi_cs_t cs)
     }
 
     if (cs == SPI_HWCS_MASK) {
-        if (spi_config[bus].cs_pin == GPIO_UNDEF) {
+        if (spi_config[bus].cs_pin == SPI_CS_UNDEF) {
```
Comment (gschorcht) [inline cpu/stm32/periph/spi.c:175]:
```suggestion
        if (gpio_is_valid(spi_config[bus].cs_pin)) {
```

File: cpu/kinetis/periph/spi.c (hunk 1 above)
Comment (aabadie) [reply cpu/kinetis/periph/spi.c:139]: This is an equal. So it should rather be `!gpio_is_valid`

Comment (gschorcht) [reply cpu/kinetis/periph/spi.c:139]: > This is an equal. So it should rather be `!gpio_is_valid`

Yeah, sure.

**Comment by aabadie** [comment]
All comments addressed and directly squashed.

**Review by gschorcht — APPROVED** [review]
Small and straightforward cleanup.

**Pull request merged**